packages/browser/src/Session.ts

```diff
@@ -157,8 +163,10 @@ export class Session extends EventEmitter {
     url: string = window.location.href
   ): Promise<ISessionInfo | undefined> => {
     if (this.info.isLoggedIn) {
+      this.fireSessionRestoreEvent();
       return this.info;
     }
```

Comment:
```suggestion
    if (this.info.isLoggedIn) {
      return this.info;
    }
```
In the browser context, if already logged in, it means no redirection happened the last login, which means there is no reason to restore the session here. This `if` test is here to prevent side-effects when calling `handleIncomingRedirect` not immediately after a redirect, with a logged in session, so I don't think we should call a callback in that context.